main.tf

```diff
@@ -1717,3 +1718,33 @@ resource "juju_integration" "sso-to-keystone" {
     endpoint = "external-idp"
   }
 }
+
+resource "juju_application" "sso-saml2-providers" {
+  for_each = var.sso-providers.saml2
+  name     = "keystone-idp-saml2-${each.key}"
+  model    = var.model
+
+  charm {
+    name     = "keystone-saml-k8s"
+    channel  = var.keystone-saml-channel
```

Comment:
Can you change the channel value to `var.keystone-saml-channel == null ? var.openstack-channel : var.keystone-saml-channel`

and in variables.tf make the default value of keystone-saml-channel to null.

For CI tests, you can pass keystone-saml-channel as tfvars with 2024.1/edge

(This makes life easier from snap-openstack point of view by just passing openstack-channel as tfvar)

Reply:
done